spec.html

```diff
@@ -5777,308 +5767,2913 @@ <h1>IterableToList ( _items_ [ , _method_ ] )</h1>
-
-        <emu-clause id="sec-object-environment-records-hasbinding-n">
-          <h1>HasBinding ( _N_ )</h1>
-          <p>The HasBinding concrete method of an object Environment Record _envRec_ takes argument _N_ (a String). It determines if its associated binding object has a property whose name is the value of the argument _N_. It performs the following steps when called:</p>
-          <emu-alg>
-            1. Let _bindings_ be the binding object for _envRec_.
-            1. Let _foundBinding_ be ? HasProperty(_bindings_, _N_).
-            1. If _foundBinding_ is *false*, return *false*.
-            1. If the _withEnvironment_ flag of _envRec_ is *false*, return *true*.
-            1. Let _unscopables_ be ? Get(_bindings_, @@unscopables).
-            1. If Type(_unscopables_) is Object, then
-              1. Let _blocked_ be ! ToBoolean(? Get(_unscopables_, _N_)).
-              1. If _blocked_ is *true*, return *false*.
-            1. Return *true*.
-          </emu-alg>
-        </emu-clause>
+<emu-clause id="sec-syntax-directed-operations">
+  <h1>Syntax-Directed Operations</h1>
+  <p>In addition to those defined in this section, specialized syntax-directed operations are defined throughout this specification.</p>
 
-        <emu-clause id="sec-object-environment-records-createmutablebinding-n-d">
-          <h1>CreateMutableBinding ( _N_, _D_ )</h1>
-          <p>The CreateMutableBinding concrete method of an object Environment Record _envRec_ takes arguments _N_ (a String) and _D_ (a Boolean). It creates in an Environment Record's associated binding object a property whose name is the String value and initializes it to the value *undefined*. If _D_ has the value *true*, the new property's [[Configurable]] attribute is set to *true*; otherwise it is set to *false*. It performs the following steps when called:</p>
-          <emu-alg>
-            1. Let _bindings_ be the binding object for _envRec_.
-            1. Return ? DefinePropertyOrThrow(_bindings_, _N_, PropertyDescriptor { [[Value]]: *undefined*, [[Writable]]: *true*, [[Enumerable]]: *true*, [[Configurable]]: _D_ }).
-          </emu-alg>
-          <emu-note>
-            <p>Normally _envRec_ will not have a binding for _N_ but if it does, the semantics of DefinePropertyOrThrow may result in an existing binding being replaced or shadowed or cause an abrupt completion to be returned.</p>
-          </emu-note>
-        </emu-clause>
+  <emu-clause id="sec-syntax-directed-operations-scope-analysis">
+    <h1>Scope Analysis</h1>
 
-        <emu-clause id="sec-object-environment-records-createimmutablebinding-n-s">
-          <h1>CreateImmutableBinding ( _N_, _S_ )</h1>
-          <p>The CreateImmutableBinding concrete method of an object Environment Record is never used within this specification.</p>
-        </emu-clause>
+    <emu-clause id="sec-static-semantics-declarationpart" type="sdo" aoid="DeclarationPart">
```

Comment:
Generally, when a clause defines an aoid, the `id` attribute is just "sec-" + lowercase(aoid).

(This is just the first occurrence of many.)

Reply:
I'm OK with this particular inconsistency, I think.